server/@types/express/index.d.ts

```diff
@@ -28,7 +28,10 @@ export declare global {
 
     interface Request {
       id: string
+      flash(type: 'errors'): ValidationError[]
       flash(type: 'errors', message: ValidationError[]): number
+      flash(type: 'formValues'): Record<string, string | string[]>[]
+      flash(type: 'formValues', message: Record<string, string | string[]>): number
```

Comment:
Subtlety here (which caused me a lot of confusion!) is that you can set `formValues` with an object,  e.g.:
 ```
req.flash('formValues', req.body) // where req.body = { visitorIds: [1, 2 ] }
```
**But**, when retrieving it you get an array:
```
const formValues = req.flash(`formValues`)
// formValues = [{ visitorIds: [1, 2 ] }]
```